lib/src/header_parser/sub_parsers/macro_parser.dart

```diff
@@ -0,0 +1,226 @@
+          clang_types.CXChildVisitResult.CXChildVisit_Break),
+      uid,
+    );
+
+    visitChildrenResultChecker(resultCode);
+    rootCursor.dispose();
+  }
+
+  // Cleanup
+  clang.clang_disposeTranslationUnit(tu);
+  clang.clang_disposeIndex(index);
+  file.deleteSync();
+
+  return _bindings;
+}
+
+/// Child visitor invoked on translationUnitCursor for parsing macroVariables.
+int _macroVariablevisitor(Pointer<clang_types.CXCursor> cursor,
+    Pointer<clang_types.CXCursor> parent, Pointer<Void> clientData) {
+  Constant constant;
+  try {
+    if (isFromGeneratedFile(cursor) &&
+        _macroVarNames.contains(cursor.spelling()) &&
+        cursor.kind() == clang_types.CXCursorKind.CXCursor_VarDecl) {
+      final e = clang.clang_Cursor_Evaluate_wrap(cursor);
+      final k = clang.clang_EvalResult_getKind(e);
+      _logger.fine('macroVariablevisitor: ${cursor.completeStringRepr()}');
+
+      /// Get macro name, the variable name starts with '<macro-name>_'.
+      final macroName = MacroVariableString.decode(cursor.spelling());
+      switch (k) {
+        case clang_types.CXEvalResultKind.CXEval_Int:
+          constant = Constant(
+            originalName: _savedMacros[macroName],
+            name: macroName,
+            rawType: 'int',
+            rawValue: clang.clang_EvalResult_getAsLongLong(e).toString(),
+          );
+          break;
+        case clang_types.CXEvalResultKind.CXEval_Float:
+          constant = Constant(
+            originalName: _savedMacros[macroName],
+            name: macroName,
+            rawType: 'double',
+            rawValue: clang.clang_EvalResult_getAsDouble(e).toString(),
+          );
+          break;
+        case clang_types.CXEvalResultKind.CXEval_StrLiteral:
+          constant = Constant(
+            originalName: _savedMacros[macroName],
+            name: macroName,
+            rawType: 'String',
+            rawValue:
+                "'${Utf8.fromUtf8(clang.clang_EvalResult_getAsStr(e).cast())}'",
+          );
+          break;
+      }
+      clang.clang_EvalResult_dispose(e);
+
+      if (constant != null) {
+        _bindings.add(constant);
+      }
+    }
+    cursor.dispose();
+    parent.dispose();
+  } catch (e, s) {
+    _logger.severe(e);
+    _logger.severe(s);
+    rethrow;
+  }
+  return clang_types.CXChildVisitResult.CXChildVisit_Continue;
+}
+
+/// Returns true if cursor is from generated file.
+bool isFromGeneratedFile(Pointer<clang_types.CXCursor> cursor) {
+  final s = cursor.sourceFileName();
+  if (s == null || s.isEmpty) {
+    return false;
+  } else {
+    return p.basename(s) == _generatedFileBaseName;
+  }
+}
+
+/// Base name of generated file.
+String _generatedFileBaseName;
+
+/// Generated macro variable names.
+///
+/// Used to determine if macro should be included in bindings or not.
+Set<String> _macroVarNames = {};
+
+/// Creates a temporary file for parsing macros in current directory.
+File createFileForMacros() {
```

Comment:
Should we move this to .dart_tool as well?

Reply:
This file is temporarily created and then deleted immediately (this happens really fast) so it doesn't matter much.

The issue with changing the location of this file is that all the headers supplied by the user will also need to be updated to use the correct path. 
Currently, the file we generate pastes the headers as it is
```hpp
#include "header1/path/pasted/as/is"
#include "header2/path/pasted/as/is"

auto abc = ABC;
auto efg = EFG;
```

Reply:
Well not really an issue, this can be done but it seems unnecessary to me. I can make this change if you want so.

Reply:
The current way sounds fine to me.